File: carbon/src/main/java/com/whatisthis/scorer/model/response/ValidationErrorResponse.java
package com.whatisthis.scorer.model.response;

import io.swagger.v3.oas.annotations.media.Schema;

import java.util.Map;

public record ValidationErrorResponse(
        @Schema(description = "Error message", example = "Invalid request body.")
        String message,
        @Schema(description = "Status code", example = "400")
        Integer status_code,
        @Schema(description = "Validation errors", example = "{\"income\":\"Income cannot be null.\",\"debt\":\"Debt cannot be negative.\"}")
        Map<String, String> errors
) {
}

File: oxygen/Loan.DTO/Response/ValidationErrorResponse.cs
using Swashbuckle.AspNetCore.Annotations;

namespace Loan.DTO.Response;

public class ValidationErrorResponse
{
    // <summary>
    // Error message
    // </summary>
    // <example>Invalid request body.</example>
    public required string Message { get; set; }
    
    // <summary>
    // Status code
    // </summary>
    // <example>400</example>
    public required int StatusCode { get; set; }
    
    // <summary>
    // Validation errors
    // </summary>
    // <example>{"loanAmount": "The LoanAmount field is required.", "term": "Term must be between 1 and 12." }</example>
    public required Dictionary<string, string> Errors { get; set; }
}